index.js

```diff
@@ -374,6 +398,125 @@ const getSecurityVulnerabilitiesForGitHubRepo = async (
   return allVulnerabilities
 }
 
+/**
+ * Get any vulnerability related to the specified Docker image using a NodeJS
+ * version that has reached End-Of-Life (EOL).
+ *
+ * @param dockerImage
+ * @param versionsCsvUrl
+ * @returns {Promise<undefined | {severity, summary, detailsUrl, ecosystem, identifiers, inspectedOn, packageName, vulnerableVersionRange}>}
+ */
+const getNodeJsVersionEolVulnerability = async (dockerImage, versionsCsvUrl) => {
+  const currentDate = getHyphenatedDate()
+  const nodeJsVersion = await docker.getNodeJsVersionOfImageUsingCache(dockerImage, versionsCsvUrl)
+  if (nodeJsVersion === 'NONE') {
+    console.log(`(NodeJS is not used in ${dockerImage})`)
+    return undefined
+  } else if (nodeJsVersion === 'UNKNOWN') {
+    const unknownDockerImageWarning = `No NodeJS version data found in spreadsheet for ${dockerImage}`
+    console.warn(`WARNING: ${unknownDockerImageWarning}`)
+    return {
+      detailsUrl: homepageOfThisLibrary + '#Missing-Docker-image-values',
+      ecosystem: null,
+      identifiers: [],
+      inspectedOn: currentDate,
+      packageName: 'NodeJS',
+      severity: 'LOW',
+      summary: unknownDockerImageWarning,
+      vulnerableVersionRange: null
+    }
+  }
+  
+  const isNodeJsVersionEOL = nodejs.isVersionEOL(nodeJsVersion, currentDate)
+  if (isNodeJsVersionEOL === undefined) {
+    const unknownNodeJsVersionWarning = `Unknown NodeJS version: ${nodeJsVersion}`
+    console.warn(`WARNING: ${unknownNodeJsVersionWarning}`)
+    return {
+      detailsUrl: homepageOfThisLibrary + '#Unknown-NodeJS-versions',
+      ecosystem: null,
+      identifiers: [],
+      inspectedOn: currentDate,
+      packageName: 'NodeJS',
+      severity: 'LOW',
+      summary: unknownNodeJsVersionWarning,
+      vulnerableVersionRange: null
+    }
+  } else if (isNodeJsVersionEOL === true) {
+    const eolNodeJsVersionWarning = `NodeJS ${nodeJsVersion} has reached End-Of-Life`
+    console.log(eolNodeJsVersionWarning)
+    return {
+      detailsUrl: 'https://nodejs.org/en/about/releases/',
+      ecosystem: null,
+      identifiers: [],
+      inspectedOn: currentDate,
+      packageName: 'NodeJS',
+      severity: 'HIGH',
+      summary: eolNodeJsVersionWarning,
+      vulnerableVersionRange: nodeJsVersion
+    }
+  }
+  
+  return undefined
+}
+
+/**
+ * Report any use of an End-Of-Life'd (EOL'd) NodeJS version in the specified
+ * Bitbucket repo.
+ *
+ * @param {string} bitbucketUsername - A Bitbucket username
+ * @param {string} bitbucketAppPassword - A Bitbucket app password
+ * @param {string} repo - A Bitbucket repo (formatted as "owner/library")
+ * @param {string} versionsCsvUrl -- The URL where we can retrieve a CSV file of
+ *     the NodeJS versions used by various Docker images.
+ * @returns {Promise<{detailsUrl, ecosystem, identifiers, inspectedOn, packageName, severity, summary, vulnerableVersionRange}[]>}
```

Comment:
I don't suppose there's a way to define a type in one place like there is with Typescript. Noting that this line re-uses the parameter list from above.

Reply:
You know, it appears that there is. I'll give it a shot (probably in a separate PR).